haystack/nodes/doc_language_classifier/base.py

```diff
@@ -0,0 +1,137 @@
+        """
+        Run language document classifier on a list of documents.
+
+        :param documents: list of documents to detect language.
+        """
+        docs_with_languages = self.predict(documents=documents)
+        output = {"documents": docs_with_languages}
+
+        if self.route_by_language is False:
+            return output, "output_1"
+
+        # self.route_by_language is True
+        languages = [doc.meta["language"] for doc in docs_with_languages]
+        unique_languages = list(set(languages))
+        if len(unique_languages) > 1:
+            raise ValueError(
+                f"If route_by_language parameter is True, Documents of multiple languages ({unique_languages}) are not allowed together. "
+                "If you want to route documents by language, you can call Pipeline.run() once for each file."
+            )
+        language = unique_languages[0]
+        if language is None:
+            logging.warning(
+                "The model cannot detect the language of any of the documents."
+                "The first language in the list of supported languages will be used to route the document: %s",
+                self.languages_to_route[0],
+            )
+            language = self.languages_to_route[0]
+        if language not in self.languages_to_route:
+            raise ValueError(
+                f"'{language}' is not in the list of languages to route ({', '.join(self.languages_to_route)})."
+                f"You should specify them when initializing the node, using the parameter languages_to_route."
+            )
+        return output, self._get_edge_from_language(str(language))
+
+    def run_batch(self, documents: List[List[Document]], batch_size: Optional[int] = None) -> Tuple[Dict, str]:  # type: ignore
+        """
+        Run language document classifier on batches of documents.
+
+        :param documents: list of lists of documents to detect language.
+        """
+        docs_lists_with_languages = self.predict_batch(documents=documents, batch_size=batch_size)
+
+        if self.route_by_language is False:
+            output = {"documents": docs_lists_with_languages}
+            return output, "output_1"
+
+        # self.route_by_language is True
+        split: Dict[str, Dict[str, List[List[Document]]]] = {
+            f"output_{pos}": {"documents": []} for pos in range(1, len(self.languages_to_route) + 1)
+        }
+
+        for docs_list in docs_lists_with_languages:
+            languages = [doc.meta["language"] for doc in docs_list]
+            unique_languages = list(set(languages))
+            if len(unique_languages) > 1:
+                raise ValueError(
+                    f"If route_by_language parameter is True, Documents of multiple languages ({unique_languages}) are not allowed together. "
+                    "If you want to route documents by language, you can call Pipeline.run() once for each file."
```

Comment:
```suggestion
                    "If you want to route documents by language, you can call Pipeline.run_batch() once for each file."
```

Reply:
(A little mistake on my part: file instead of Document)

I understand the correction, but I think `run_batch` should be called on batches of Documents.
So I wouldn't suggest to call `run_batch` once for each document...
WDYT?